recipes/aws-libfabric/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class LibfabricConan(ConanFile):
+    name = "aws-libfabric"
+    description = "AWS Libfabric"
+    topics = ("fabric", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/libfabric"
+    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ["gni", "psm", "psm2", "sockets", "rxm", "tcp", "udp", "usnic", "verbs", "bgq", "shm", "efa", "rxd", "mrail", "rstream", "perf", "hook_debug"]
+    options = {
+        **{ p: [True, False, "shared"] for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": [True, False],
+            "with_bgq_progress": ["auto", "manual"],
+            "with_bgq_mr": ["basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: False for p in _providers },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "tcp": True,
+            "with_libnl": False,
+            "with_bgq_progress": "manual",
+            "with_bgq_mr": "basic"
+        }
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        elif self.settings.os == "Linux":
+            self.options.efa = True
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_libnl:
+            self.requires("libnl/3.2.25")
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The libfabric package cannot be built on Visual Studio.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        with tools.chdir(self._source_subfolder):
+            self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
```

Comment:
Just run `autoreconf` directly.
It gives us more assurance that the `autogen.sh` script is not sneakily downloading anything.
```suggestion
            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
```